docs/explanation/priority-inversion-inheritance.rst

```diff
@@ -0,0 +1,67 @@
+Priority inversion and inheritance
+==================================
+
+Among the reasons why PREEMPT_RT is not in mainline yet is that much of the locking within the kernel has to be updated to prevent priority inversion from occurring in a real-time environment.
+The present and the following section will introduce unbounded priority inversion and the need for priority inheritance.
+
+Unbounded priority inversion
+----------------------------
+
+Let's begin with priority inversion by looking at the diagram below.
+Three tasks, L, M, and H, with varying priority levels, low, medium and high, are present in the kernel and about to contest for CPU access.
+
+The low-priority task L runs until it takes a lock; in the diagram below, the blue bar turns purple.
+After acquiring it, task L holds the lock and begins modifying some critical sections within the kernel. 
+Once the higher-priority task H appears, it preempts task L and starts to run.
+At this point, task H would like to acquire the same lock task L is holding.
+As it can't do so, the higher-priority task H goes to sleep and waits for the lower-priority task L to release the lock.
+Task L thus continues running while Task H is sleeping.
+In such a setup, priority inversion can occur if a medium-priority task M comes along and preempts task L.
+Once task M starts running, the high-priority task H will potentially wait for an unbounded amount of time, preventing it from doing work in a critical kernel section.
+Improving the flexibility to preempt tasks executing within the kernel would thus help guarantee an upper time boundary.
+
+.. figure:: priority-inversion-inheritance/unbounded-priority-inversion.svg
+    :width: 100%
+    :align: center
+    :alt: Unbounded priority inversion
+
+    Unbounded priority inversion: A high-priority task being indefinitely delayed by a medium-priority task due to a lack of proper priority management.
```

Comment:
By the way, this isn't part of this PR's scope.
But we might want to make some changes to the CSS for captions. I think the caption text should look a little different from normal paragraph text.
Personal preference but calling this out anyway.